### PR TITLE
Add AdamW Weight Decay Back As Advanced Parameter

### PR DESCRIPTION
**Pull request by ExponentialML**
Generally speaking, this is the only *"meaningful"*  hyperparameter to modify when using AdamW (Adam does a very good job of doing this for us).
It can be used in various ways, but a *"good"* example use case is setting a good learning rate (example 1e-4) and just tuning the weight decay. 

Per the tooltip (simplified): 

> _"AdamW Weight Decay": "The weight decay of the AdamW Optimizer. Values closer to 0 closely match your training dataset, and values closer to 1 generalize more and deviate from your training dataset. Default is 1e-2, values lower than 0.1 are recommended."_

References:
_[Understanding AdamW: Weight decay or L2 regularization?](https://www.fast.ai/posts/2018-07-02-adam-weight-decay.html#adamw)_
_[Can weight decay be higher than learning rate](https://stats.stackexchange.com/questions/324616/can-weight-decay-be-higher-than-learning-rate)_
_[Deep learning basics — weight decay](https://medium.com/analytics-vidhya/deep-learning-basics-weight-decay-3c68eb4344e9)_

Builds off of commit: https://github.com/d8ahazard/sd_dreambooth_extension/commit/bde92f19a556116b4b7f869158ef6e20945742bf

### CODE REVIEW

**Comment by d8ahazard** [comment]
Beautiful work, as always. Could you kindly also add a blurb to the readme with some of the info you referenced. Maybe I'll start messing with this as well.

**Comment by ExponentialML** [comment]
> Beautiful work, as always. Could you kindly also add a blurb to the readme with some of the info you referenced. Maybe I'll start messing with this as well.

Thanks, and sure! We should make sure it works properly first before merging. If the params aren't loading / saving properly, it could skew traning results pretty drastically and could be a culprit for issues arising.

Testing a value of 1 should result in extreme generalization, while the opposite should overfit the training data. The default should always be `1e-2` as per the [documentation](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html).
